__tests__/app.test.js

```diff
@@ -84,268 +84,477 @@ describe("GET /api/articles", () => {
+          const { articles } = returnedPromises[i].body;
+
+          expect(articles).toBeSortedBy(validSortBys[i], { descending: true });
+        }
+      });
+    });
+    describe("article objects are filtered by valid topic", () => {
+      test("filtered by mitch", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(11);
+            articles.forEach((article) => {
+              expect(article).toEqual(
+                expect.objectContaining({ topic: "mitch" })
+              );
+            });
+          });
+      });
+      test("filtered by cats", () => {
+        return request(app)
+          .get("/api/articles?topic=cats")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(1);
+            expect(articles[0]).toEqual(
+              expect.objectContaining({
+                article_id: 5,
+                title: "UNCOVERED: catspiracy to bring down democracy",
+                topic: "cats",
+                author: "rogersop",
+                body: "Bastet walks amongst us, and the cats are taking arms!",
+                created_at: "2020-08-03T13:14:00.000Z",
+                votes: 0,
+                comment_count: 2,
+              })
+            );
+          });
+      });
+      test("filtered by paper", () => {
+        return request(app)
+          .get("/api/articles?topic=paper")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(0);
+          });
+      });
+    });
+    test("400: responds with 'Bad request: invalid sort_by'", () => {
       return request(app)
-        .patch("/api/articles/numberone")
-        .send(requestBody)
+        .get("/api/articles?sort_by=body")
         .expect(400)
         .then(({ body: { message } }) => {
-          expect(message).toBe("Bad request");
+          expect(message).toBe("Bad request: invalid sort_by");
         });
     });
-    test("when request body is empty", () => {
+    test("400: responds with 'Bad request: invalid order'", () => {
       return request(app)
-        .patch("/api/articles/1")
-        .send()
+        .get("/api/articles?sort_by=title&order=toptobottom")
         .expect(400)
         .then(({ body: { message } }) => {
-          expect(message).toBe("Bad request: request body of invalid format");
+          expect(message).toBe("Bad request: invalid order");
         });
     });
-    test("when request body is of wrong format", () => {
-      const requestBody = {
-        inc_votes: "apple",
-      };
-
+    test("400: responds with 'Bad requestL invalid topic'", () => {
       return request(app)
-        .patch("/api/articles/1")
-        .send(requestBody)
+        .get("/api/articles?sort_by=title&order=ASC&topic=meech")
```

Comment:
This should be a 404 error message as it's valid syntax for a topic but not found

Reply:
changed the sort_by and topic errors to 404 from 400.